src/views/clusteroverview/OverviewTab/running-vms-per-template-card/hooks/useVMsPerTemplateResources.ts

```diff
@@ -0,0 +1,54 @@
+import { useMemo } from 'react';
+
+import {
+  modelToGroupVersionKind,
+  TemplateModel,
+  VirtualMachineModelGroupVersionKind,
+} from '@kubevirt-ui/kubevirt-api/console';
+import { TEMPLATE_TYPE_LABEL } from '@kubevirt-utils/resources/template';
+import { K8sResourceCommon, useK8sWatchResources } from '@openshift-console/dynamic-plugin-sdk';
+
+type UseVMsPerTemplateResources = (namespace: string) => {
+  loaded: boolean;
+  vms: K8sResourceCommon[];
+  templates: K8sResourceCommon[];
+};
+
+const useVMsPerTemplateResources: UseVMsPerTemplateResources = (namespace) => {
+  const watchedResources = {
+    vms: {
+      groupVersionKind: VirtualMachineModelGroupVersionKind,
+      isList: true,
+      namespaced: !!namespace,
+      namespace,
+    },
+    templates: {
+      groupVersionKind: modelToGroupVersionKind(TemplateModel),
+      isList: true,
+      namespaced: !!namespace,
+      namespace,
+      selector: {
+        matchExpressions: [
+          {
+            key: TEMPLATE_TYPE_LABEL,
+            operator: 'Exists',
+          },
+        ],
+      },
+    },
+  };
+
+  const resourceData = useK8sWatchResources<{ [key: string]: K8sResourceCommon[] }>(
+    watchedResources,
+  );
+
+  return useMemo(() => {
+    return {
+      templates: resourceData.templates.data,
+      vms: resourceData.vms.data,
+      loaded: resourceData.templates.loaded && resourceData.vms.loaded,
```

Comment:
please add ?. , im not sure if those nested props are guaranteed using watchResourceS

Reply:
Done!